bundle/libraries/libraries.go

```diff
@@ -92,13 +93,13 @@ func findArtifactsAndMarkForUpload(ctx context.Context, lib *compute.Library, b
 	}
 
 	if len(matches) == 0 && isLocalLibrary(lib) {
-		return fmt.Errorf("no library found for %s", libPath(lib))
+		return fmt.Errorf("file %s is referenced in libraries section but no such file found on local file system", libPath(lib))
```

Comment:
```suggestion
		return fmt.Errorf("file %s is referenced in libraries section but doesn't exist on the local file system", libPath(lib))
```

Reply:
I think this only happens as part of the deploy, but I suspect we can check that this is all valid during validation if we don't require a build step, right?

Reply:
`libraries.MatchWithArtifacts` must happen after artifact build and infer steps. Otherwise logic in MatchWithArtifacts becomes to complex and conditional, for example, is there no file locally because it was not yet build, or it really does not exists, or it wasn't infered.

The preferred flow is 
1. Read the config and artifacts section
2. Infer anything that is missing and we can infer
3. Build artifacts based on the result configuration
4. Take the result artifacts configuration and state of the file system and see if libraries used matches the one on FS.

Reply:
Thanks -- we should capture this somewhere.